app/Http/Controllers/SquareController.php

```diff
@@ -0,0 +1,206 @@
+            $payment->method = 'square';
+            $payment->recorded_by = $user->id;
+            $payment->unique_id = self::generateUniqueId();
+            $payment->notes = 'Checkout flow started';
+
+            $transaction->payment()->save($payment);
+        } else {
+            return view(
+                'square.error',
+                [
+                    'message' => 'We could not find a transaction ready for payment.',
+                ]
+            );
+        }
+
+        $amount = (int) ($transaction->package->cost * 100);
+
+        $basePrice = new Money();
+        $basePrice->setAmount($amount);
+        $basePrice->setCurrency('USD');
+
+        $surcharge = new Money();
+        $surcharge->setAmount(self::calculateSurcharge($amount));
+        $surcharge->setCurrency('USD');
+
+        $orderLineItem = new OrderLineItem('1');
+        $orderLineItem->setName('Dues - '.$transaction->package->name);
+        $orderLineItem->setBasePriceMoney($basePrice);
+
+        $orderServiceCharge = new OrderServiceCharge();
+        $orderServiceCharge->setName('Card Processing Surcharge');
+        $orderServiceCharge->setAmountMoney($surcharge);
+        $orderServiceCharge->setCalculationPhase(OrderServiceChargeCalculationPhase::TOTAL_PHASE);
+
+        $order = new Order(config('square.location_id'));
+        $order->setReferenceId((string) $payment->id);
+        $order->setLineItems([$orderLineItem]);
+        $order->setServiceCharges([$orderServiceCharge]);
+
+        $orderRequest = new CreateOrderRequest();
+        $orderRequest->setOrder($order);
+        $orderRequest->setLocationId(config('square.location_id'));
+        $orderRequest->setIdempotencyKey($payment->unique_id);
+
+        $checkoutRequest = new CreateCheckoutRequest($payment->unique_id, $orderRequest);
+        $checkoutRequest->setMerchantSupportEmail('treasurer@robojackets.org');
+        $checkoutRequest->setPrePopulateBuyerEmail($user->gt_email);
+        $checkoutRequest->setRedirectUrl(route('pay.complete'));
+
+        $square = new SquareClient([
+            'accessToken' => config('square.access_token'),
+            'environment' => config('square.environment'),
+        ]);
+
+        $checkoutApi = $square->getCheckoutApi();
+        $checkoutResponse = $checkoutApi->createCheckout(config('square.location_id'), $checkoutRequest);
+
+        if (! $checkoutResponse->isSuccess()) {
+            Log::error(self::class.' Error creating checkout - '.json_encode($checkoutResponse->getErrors()));
+
+            return view(
+                'square.error',
+                [
+                    'message' => 'We could not contact Square to begin your payment.',
+                ]
+            );
+        }
+
+        $checkout = $checkoutResponse->getResult()->getCheckout();
+
+        $payment->checkout_id = $checkout->getId();
+        $payment->order_id = $checkout->getOrder()->getId();
+        $payment->save();
```

Comment:
The `checkout_id` is used to verify the redirect and the `order_id` is used to match up webhook calls. The rest of the IDs in the database aren't necessary anymore (at least for web payments, card-present may still need them?)